src/Temp.c

```diff
@@ -0,0 +1,176 @@
+/// @file my-project.c
+#include <stdio.h>
+#include "libopencm3/stm32/rcc.h"
+#include "libopencm3/stm32/gpio.h"
+#include "libopencm3/stm32/i2c.h"
+#include "libopencm3/stm32/usart.h"
+
+#define I2C1_CLCK_MHZ					(1000000UL)		/// Used in calculation of the I2C1 timing
+#define I2C1_SCL_PIN					(GPIO6)			/// Pin definition for SCL of I2C1
+#define I2C1_SDA_PIN					(GPIO9)			/// Pin definition for SDA of I2C1
+#define TEMP_SENS_ADDR					(0xEE)			/// Address of the temperature sensor
+#define TEMP_REG						(0x2E)			
+#define PRES_REG			 			(0X34)
+
+#define USART_PORT						(USART1)
+#define USART_BAUDRATE					(115200)
+#define USART1_TX_PIN					(GPIO9)			
+#define USART1_RX_PIN					(GPIO10)	
+#define USART_WORD_LEN					(8)			
+
+/**
+ * Delay
+*/
+static void delay(void)
+{
+	for(int i = 0; i < 250000; i++);
```

Comment:
This is a valid way to make a delay, yes.
but it is not a good way to do it.
In this way we block the execution and cycle.
This might not be even correct as time delay, because when you have multiple interrupts during the function execution the time taken in the interrupts will not be accounted for in the loop.

a better more accurate way to do it is with a timer in case you need higher precision, or with looking at the system timer counter (if any) when you don't need high precision,  but time sync over time will be accurate.